src/main/java/se/kth/hopsworks/rest/DataSetService.java

```diff
@@ -11,6 +11,7 @@
 import javax.ejb.TransactionAttribute;
```

Comment:
This class contains too much functionality. All the methods should do is call a controller, which implements the logic, and handle any possible errors for display in the frontend. That way we separate the frontend displaying from the backend logic.
Look at other Service classes in this package for a guideline. (Except for ProjectService, which suffers from the same issue.)